20181204-Caroline.java

```diff
@@ -1,17 +1,17 @@
 Past paper 2015
 main idea: String, class
 public class HiddenWord{
-    private String realWord;
+    private String puzzle;
     public HiddenWord(String s1){
-        realWord=s1;
+        puzzle=s1;
     }
     public String getHint(String guess){
     String temp="";
     for(int i=0;i<guess.length();i++){
         if(realWord.subString(i,i+1).equals(guess.subString(i,i+1)){
```

Comment:
But when you change the name of variable realword, you ignore the line 11. Be careful next time.